docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,490 @@
+        ))}
+      </div>
+    );
+  }}
+```
+
+Later, you write a component for subscribing to a single blog post, which follows a similar pattern:
+
+```js
+class BlogPost extends React.Component {
+  constructor(props) {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {
+      blogPost: DataSource.getBlogPost(props.id)
+    };
+  }
+
+  componentDidMount() {
+    DataSource.addChangeListener(this.handleChange);
+  }
+
+  componentWillUnmount() {
+    DataSource.removeChangeListener(this.handleChange);
+  }
+
+  handleChange() {
+    this.setState({
+      blogPost: DataSource.getBlogPost(this.props.id)
+    });
+  }
+
+  render() {
+    return <BlogPost blogPost={this.state.blogPost} />;
+  }}
+```
+
+`CommentList` and `BlogPost` aren't identical—they call different methods on `DataSource`, and they render different output. But much of their implementation is the same:
+
+- On mount, add a change listener to `DataSource`.
+- Inside the listener, call `setState` whenever the data source changes.
+- On unmount, remove the change listener.
+
+You can imagine that in a large app, this same pattern of subscribing to `DataSource` and calling `setState` will occur over and over again. We want an abstraction that allows us to define this logic in a single place and share them across many components. This is where HOCs excel.
+
+We'll create a function called `withSubscription` that works like this:
+
+```js
+const CommentListWithSubscription = withSubscription(
+  // The first parameter is the wrapped component
+  CommentList,
+  // The second parameter retrieves the data we're interested in
+  (DataSource) => DataSource.getComments()
+);
+
+const BlogPostWithSubscription = withSubscription(
+  CommentList,
+  // An optional second argument passes the current props
+  (DataSource, props) => DataSource.getBlogPost(props.id)
+});
+```
+
+And here's what the implementation of `withSubscription` looks like:
+
+```js
+// This function takes a component...
+function withSubscription(WrappedComponent, selectData) {
+  // ...and returns another component...
+  return class extends React.Component {
+    constructor(props) {
+      super();
+      this.handleChange = this.handleChange.bind(this);
+      this.state = {
+        data: selectData(DataSource, props)
+      };
+    }
+
+    componentDidMount() {
+      // ... that takes care of the subscription...
+      DataSource.addChangeListener(this.handleChange);
+    }
+
+    componentWillUnmount() {
+      DataSource.removeChangeListener(this.handleChange);
+    }
+
+    handleChange() {
+      this.setState({
+        comments: selectData(DataSource, this.props)
+      });
+    }
+
+    render: function() {
+      // ... and renders the wrapped component with the fresh data!
+      // Notice that we pass through any additional props
+      return <WrappedComponent data={this.state.data} {...this.props} />;
+    }
+  });
+}
+```
```

Comment:
Maybe add "Note that the returned component does not extend the wrapped component. This is not an example of inheritance. Instead, it uses the wrapped component in the `render()` method."